### DISCUSSION TITLE
scala 3.2.2

### DISCUSSION THREAD

**Discussion by firozahamed05** [scalameta/metals#5195] [General]
How I get scala 3.2.2 ?

**Comment by ckipp01** [comment]
Hi @firozahamed05 the Scala version for you project is typically set via your build tool. For example if you're using sbt you'd want to just make sure you `scalaVersion` is set to `3.2.2`.

**Reply by firozahamed05** [reply]
I want to download it.

**Reply by ckipp01** [reply]
I'm not sure I understand your question in relation to Metals. Are you trying to download that version of Scala to use the REPL? Or are you editing a project that you'd like to use Scala 3 with?

**Reply by firozahamed05** [reply]
yes !

**Reply by ckipp01** [reply]
> yes !

which one?

**Reply by firozahamed05** [reply]
editing a project that I'd like to use Scala 3 with

**Reply by ckipp01** [reply]
What build tool does your project use?

**Reply by firozahamed05** [reply]
spark

**Reply by ckipp01** [reply]
Spark is not a build too. I recommend jumping onto the [Scala Discord](https://discord.gg/scala) and asking your questions there as they seem more general than Metals.

**Reply by firozahamed05** [reply]
Sorry dear kipp,
actually I hava extended scala(metals) in my VSCODE. But when I run code,it says no build target found for path and i can't fix it. would you help me.

**Reply by ckipp01** [reply]
As I asked before

> What build tool does your project use?

sbt, Maven, Gradle, something else?

You should have been prompted to import your build when you opened the project in VS Code, did this happen?

**Reply by firozahamed05** [reply]
sbt and yes

**Reply by ckipp01** [reply]
and did the import succeed? Did you get a warning? What the the Metals Doctor show you?

**Reply by firozahamed05** [reply]
Chris Kipp,
Now it's working.Thanks for your company.
 May I discuss with you if arrise  any problem  later ?

**Reply by tgodzik** [reply]
You can always let us know via issues, discussions or discord if you have any issues. We will try to help out, but we cannot promise any availability.

**Reply by firozahamed05** [reply]
val result = (  a:Int, b :Int)=> {
      while(a!=b){ 
     if (a>b) a-=b  else  b-=a
     }
       return a
     }

I can't solve this problem.

**Reply by tgodzik** [reply]
you would need to be more exact than that. An exact snippet of code and what doesn't work

**Reply by firozahamed05** [reply]
if(a>b) a-=b else b-=a
}
      return a
}

**Reply by ckipp01** [reply]
Hey @firozahamed05 I think it might be best to ask specific Scala syntax or other generic Scala questions in a more appropriate place like the [Scala Discord](https://discord.gg/scala). I'm going to go ahead and lock this seeing that these questions are sort of "out of scope" for Metals. If you do have any issues getting Metals working or encounter any bugs, please do open another discussion/issue.
